### DISCUSSION TITLE
@next/font is not always loading fonts

### DISCUSSION THREAD

**Discussion by M1ck0** [vercel/next.js#44864] [Help] [answered]
https://user-images.githubusercontent.com/19279947/212365304-afa21f7c-0094-40cc-a19e-8919fd9809a5.mov

This happens when the page reloads. Sometimes the font is loaded, and sometimes it's not. 

Here is the code.

```jsx
import { Nunito } from "@next/font/google";

const nunito = Nunito({ subsets: ["latin"], weight: ["400", "500", "600"] });

const Layout = ({ children }) => {
  return (
    <>
      <main className={"main" + " " + nunito.className}>
        <Header />
        {children}
        <Footer />
      </main>
    </>
  );
};
```
Any idea why is this happening? I am on the latest `NextJS` version.

**Comment by JamieGrimwood** [comment]
Same issue using barlow font. No idea why it happens.

**Reply by abrambailey** [reply]
We are having the same issue on HearingTracker.com

**Reply by ChinonsoIg** [reply]
For your case, Next docs recommends we use [variable fonts](https://fonts.google.com/variablefonts). Barlow font is not on this list.
Read more on the docs [here](https://nextjs.org/docs/basic-features/font-optimization).

@JamieGrimwood

**Reply by M1ck0** [reply]
@ChinonsoIg I am using variable font and still have the issue

**Reply by ChinonsoIg** [reply]
@M1ck0  Yours is most probably related to network. If the Google font takes time to load, the fallback font is then used.

**Reply by JamieGrimwood** [reply]
> For your case, Next docs recommends we use [variable fonts](https://fonts.google.com/variablefonts). Barlow font is not on this list.
> 
> Read more on the docs [here](https://nextjs.org/docs/basic-features/font-optimization).
> 
> 
> 
> @JamieGrimwood 

It works absolutely fine when I set the option to swap, so not sure why the variable font thing is needed.

**Comment by bdeterling** [comment]
I may have figured this out. The document says the default font display is "swap", but I left it off and I'm seeing it come in as optional, which says "Gives the font face an extremely small block period and no swap period." That would mean it tries to load the font, but if it's not fast enough, it falls back and never swaps. I changed it to be display: swap; and now, every once in a while, I see the fallback font, but it immediately ... swaps.

**Reply by M1ck0** [reply]
Do you mean you set `display: swap` in CSS globally, or in `@next/font` config?

**Reply by ChinonsoIg** [reply]
I guess it's something like below, where **Inter** is the font imported from '@next/font/google'
```
const inter = Inter({
  weight: '400',
  display: 'swap',
  subsets: ['latin'],
});
```

**Comment by bdeterling** [comment]
In the @next/font config. The documentation says swap is the default, but that does not seem to be the case, at least with 13.0.4.

**Reply by JamieGrimwood** [reply]
> In the @next/font config. The documentation says swap is the default, but that does not seem to be the case, at least with 13.0.4.

This was my exact problem. Setting the option to swap fixed it.

**Comment by bdeterling** [comment, marked as answer]
Looks like it was fixed two weeks ago:

https://github.com/vercel/next.js/pull/45430
